packages/slice-machine/src/components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -1,35 +1,62 @@
 import { Text } from "@prismicio/editor-ui";
-import { FC } from "react";
+import clsx from "clsx";
+import {
+  createContext,
+  PropsWithChildren,
+  ReactNode,
+  useContext,
+  useMemo,
+} from "react";
 
-import * as styles from "./Breadcrumb.css";
+import styles from "./Breadcrumb.module.css";
 
-export type BreadcrumbProps = {
-  folder: string;
-  page?: string;
-  separator?: string;
+type BreadcrumbContext = {
+  separator?: ReactNode;
 };
 
-export const Breadcrumb: FC<BreadcrumbProps> = ({
-  folder,
-  page,
-  separator = "/",
-}) => {
+const BreadcrumbContext = createContext<BreadcrumbContext>({});
+
+type BreadcrumbProps = PropsWithChildren<{
+  separator?: ReactNode;
+}>;
+
+export function Breadcrumb(props: BreadcrumbProps) {
+  const { separator = "/", children, ...restProps } = props;
+  const contextValue = useMemo(() => ({ separator }), [separator]);
+
   return (
-    <div
-      aria-label="Breadcrumb"
-      data-testid={`breadcrumb-${folder}-${page ?? ""}`}
-    >
-      <Text color="grey11">
-        {folder}
-        {page !== undefined ? (
-          <>
-            &nbsp;{separator}&nbsp;
-            <Text className={styles.pageSpan} component="span" color="grey12">
-              {page}
-            </Text>
-          </>
-        ) : null}
+    <nav aria-label="Breadcrumb" {...restProps}>
+      <ol className={styles.items}>
+        <BreadcrumbContext.Provider value={contextValue}>
+          {children}
+        </BreadcrumbContext.Provider>
+      </ol>
+    </nav>
+  );
+}
+
+type BreadcrumbItemProps = PropsWithChildren<{
+  active?: boolean;
+}>;
+
+export function BreadcrumbItem(props: BreadcrumbItemProps) {
+  const { active = false, children, ...otherProps } = props;
+  const { separator } = useContext(BreadcrumbContext);
+
+  return (
+    <li {...otherProps} className={styles.item}>
+      <Text
+        component="span"
+        color={active ? "grey12" : "grey11"}
+        className={clsx(active && styles.active)}
+      >
+        {children}
       </Text>
-    </div>
+      <div role="presentation" className={styles.separator}>
+        <Text component="span" color="grey11">
```

Comment:
_✍️ asking for a potential change_

I see that we had a style issue problem before, what I see in the Figma design is that the separator is in `grey12`. Maybe @benjammartin you could confirm what I'm saying? ([see here, Breadcrumb component](url))

Reply:
Second mistake. It's grey11.

Reply:
Thank you again! So no need to change here also 🙂